tests/e2e_logging.py

```diff
@@ -1331,13 +1344,23 @@ def run(args):
 
 
 if __name__ == "__main__":
-
-    args = infra.e2e_args.cli_args()
-    if args.js_app_bundle:
-        args.package = "libjs_generic"
-    else:
-        args.package = "liblogging"
-    args.nodes = infra.e2e_args.max_nodes(args, f=0)
-    args.initial_user_count = 4
-    args.initial_member_count = 2
-    run(args)
+    jsargs = infra.e2e_args.cli_args()
+    label = jsargs.label
+
+    jsargs.package = "libjs_generic"
+    jsargs.nodes = infra.e2e_args.max_nodes(jsargs, f=0)
+    jsargs.initial_user_count = 4
+    jsargs.initial_member_count = 2
+    jsargs.label = f"js_{label}"
+
+    cppargs = copy.deepcopy(jsargs)
+    cppargs.js_app_bundle = None
+    cppargs.package = "liblogging"
+    jsargs.label = f"cpp_{label}"
+
+    js = threading.Thread(name="JS", target=run, args=[jsargs])
+    cpp = threading.Thread(name="CPP", target=run, args=[cppargs])
+    js.start()
+    cpp.start()
+    js.join()
+    cpp.join()
```

Comment:
I think we should have CLI arguments to run just one of these, to simplify debugging.

Reply:
Yes, that's a good idea, and also to list all the available ones probably. I think we also want an easy mapping from the handle name to the label too, so we can find the directories easily.

So how about:

$TEST_NAME := name of test in CTest
$SUBTEST_NAME := name of Python thread running a portion of a test
$LABEL := $TEST_NAME + "_" + $SUBTEST_NAME

./test.py -N           list all sub-tests in test
./test.py -R $STR   run only sub-tests whose name includes $STR

That's very basic but consistent with CTest and easy to use for now.

Reply:
Sounds good to me.